### PR TITLE
Migrate code from managed-tenants GitLab

### PR DESCRIPTION
**Pull request by mayankshah1607**
Signed-off-by: Mayank Shah <m.shah@redhat.com>

### CODE REVIEW

**Comment by sugarraysam** [comment]
Thank you for this PR @mayankshah1607 .

Couple things we probably want here:
- `Makefile`
- `.pre-commit-config.yaml`
- `tests/` folder
- `requirements-dev.txt`
- `.linters/` folder
- Copy the initial `.gitignore`

WDYT?

**Comment by mayankshah1607** [comment]
Thanks for the feedback @sugarraysam !

> The only thing is we need to remove tests/addons/test_pre_tasks.py

I haven't added the `tests/` directory in this PR as I wasn't yet sure of how much of it is required in the `managed-tenants` CI. The plan was to add those as a follow-up PR.

> I can also do the work in a new PR, as it might require some minor tweaks. My goal was to get `make test` to work.

Sounds good to me. Please feel free to pick this up :)

**Comment by mayankshah1607** [comment]
> Meanwhile, I'm adding the files/folders included in your list here :)

Done! Please take a look and feel free to merge once it looks good :)

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import os
+
+from setuptools import setup, find_packages
+
+
+BASE_PATH = os.path.dirname(__file__)
+
+
+def get_readme_content():
+    with open(os.path.join(BASE_PATH, 'README.rst'), 'r') as readme:
```
Comment (sugarraysam) [inline setup.py:24]: `$ make develop` fails here needs to be `README.md`

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        return readme.read()
+
+
+def get_version():
+    with open(os.path.join(BASE_PATH, 'VERSION'), 'r') as version:
```
Comment (sugarraysam) [inline setup.py:29]: `$ make develop` fails here we need this `VERSION` file after all.

**Review by sugarraysam — APPROVED** [review]
Looks good from my side, just these two minor tweaks and feel free to merge.

**Pull request merged**